File: hnaj-be/app/Http/Controllers/Api/Auth/LogoutController.php
<?php

namespace App\Http\Controllers\Api\Auth;

use App\Actions\Auth\RevokeAccessTokens;
use App\Http\Controllers\Controller;
use App\Http\Responses\ApiResponse;
use Illuminate\Http\JsonResponse;
use Illuminate\Http\Request;

class LogoutController extends Controller
{
    public function __invoke(Request $request, RevokeAccessTokens $revokeAccessTokens): JsonResponse
    {
        $revokeAccessTokens->handleCurrent($request->user());

        return ApiResponse::success(message: 'Signed out successfully.');
    }
}

File: hnaj-be-java/src/main/java/com/hnaj/auth/controller/LogoutController.java
package com.hnaj.auth.controller;

import com.hnaj.api.common.ApiResponse;
import com.hnaj.auth.AccessTokenService;
import com.hnaj.auth.AuthenticatedUser;
import java.util.Map;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

/** POST /api/auth/logout — bearer only, no role middleware (routes/api.php line 101). */
@RestController
public class LogoutController {
    private final AccessTokenService accessTokens;

    public LogoutController(AccessTokenService accessTokens) {
        this.accessTokens = accessTokens;
    }

    @PostMapping("/api/auth/logout")
    public ResponseEntity<Map<String, Object>> logout(AuthenticatedUser principal) {
        accessTokens.revoke(principal.rawToken());
        return ApiResponse.success(null, "Signed out successfully.", null);
    }
}